File: translate.py
import os
from openai import OpenAI

client = OpenAI(
# Set up your OpenAI API key
api_key = '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',
)

def translate_text(text, target_language="Swedish"):

    # Use the Completion.create method for text-based completion
    response = client.chat.completions.create(
        model="gpt-3.5-turbo",  # You can change this model if needed
        messages=[
        {
            "role": "user",
            "content": "Translate the following English text to swedish: \n\n{text}\n\n"
        }
    ],
        max_tokens=100000,
        temperature=0.5,
    )
    
    translated_text = response.choices[0].text.strip()
    return translated_text

def translate_file(file_path, target_language="Swedish"):
    with open(file_path, 'r', encoding='utf-8') as file:
        content = file.read()
    
    translated_content = translate_text(content, target_language)
    
    # Write the translated content to a new file
    output_file_path = file_path.replace('.md', f'_{target_language.lower()}.md')
    with open(output_file_path, 'w', encoding='utf-8') as file:
        file.write(translated_content)

    print(f"Translated {file_path} to {output_file_path}")

# Recursively translate all markdown files in the docs directory
for subdir, dirs, files in os.walk('docs'):
    for file in files:
        if file.endswith('.md'):
            translate_file(os.path.join(subdir, file))

File: translate.js
import fs from 'fs';
import path from 'path';
import OpenAIApi from 'openai';
import Configuration from 'openai';


// Set up your OpenAI API key from environment variables
const configuration = new Configuration({
  apiKey: process.env.OPENAI_API_KEY,
});
const openai = new OpenAIApi(configuration);

// Function to translate text using OpenAI API
const translateText = async (text, targetLanguage = "Swedish") => {
  const prompt = `Translate the following English text to ${targetLanguage}:\n\n${text}`;

  try {
    const response = await openai.chat.completions.create({
    model: "gpt-3.5-turbo", 
    messages: [
      { role: "system", content: "You are a helpful assistant that translates text." },
      { role: "user", content: "translate following text to swedish" }
    ],
    max_tokens: 4000,
    temperature: 0.5,
  });

  if (response && response.data && response.data.choices && response.data.choices.length > 0) {
      return response.data.choices[0].text.trim();
    } else {
      console.error("OpenAI   
 API call failed or unexpected response format:", response);
      return ""; // Handle empty response or error
    }
  } catch (error) {
    console.error("Error during OpenAI API call:", error);
    return ""; // Handle API call error
  }
};

// Function to translate a file and save the translated content
const translateFile = async (filePath, targetLanguage = "Swedish") => {
  const content = fs.readFileSync(filePath, 'utf-8');
  const translatedContent = await translateText(content, targetLanguage);

  const outputFilePath = filePath.replace('.md', `_${targetLanguage.toLowerCase()}.md`);
  fs.writeFileSync(outputFilePath, translatedContent, 'utf-8');

  console.log(`Translated ${filePath} to ${outputFilePath}`);
};

// Recursively translate all Markdown files in the docs directory
const docsDir = path.join(process.cwd(), 'docs');
const files = fs.readdirSync(docsDir);

files.forEach(async (file) => {
  const filePath = path.join(docsDir, file);
  if (file.endsWith('.md')) {
    await translateFile(filePath);
  }
});
